include/extractor/guidance/toolkit.hpp

```diff
@@ -370,9 +389,39 @@ inline bool requiresNameAnnounced(const std::string &from, const std::string &to
         (from_ref.find(to_ref) != std::string::npos || to_ref.find(from_ref) != std::string::npos);
     const auto ref_is_removed = !from_ref.empty() && to_ref.empty();
 
-    const auto obvious_change =
-        (names_are_empty && refs_are_empty) || (names_are_equal && ref_is_contained) ||
-        (names_are_equal && refs_are_empty) || name_is_removed || ref_is_removed;
+    const auto checkForSuffixChange = [](const std::size_t common_length, const std::string &first,
+                                         const std::string &second,
+                                         const SuffixTable &suffix_table) {
+        if (0 == common_length)
+            return false;
+
+        std::string first_suffix =
+            first.size() > common_length
+                ? first.substr(common_length + (first[common_length] == ' ' ? 1 : 0))
+                : "";
+        boost::algorithm::to_lower(first_suffix);
+
+        std::string second_suffix =
+            second.size() > common_length
+                ? second.substr(common_length + (second[common_length] == ' ' ? 1 : 0))
+                : "";
+        boost::algorithm::to_lower(second_suffix);
+
+        const auto first_delta_is_suffix =
+            first_suffix.empty() || suffix_table.isSuffix(first_suffix);
+        const auto second_delta_is_suffix =
+            second_suffix.empty() || suffix_table.isSuffix(second_suffix);
+
+        return first_delta_is_suffix && second_delta_is_suffix;
+    };
```

Comment:
This lambda is a bit ugly, and has dup. code for fist / second suffix. Maybe refactor a bit.